### ISSUE TITLE
Uploading attachments broken in 1.0.4

### ISSUE THREAD

**Issue by finchr** [iamjackg/md2cf#18]
I get the following error when I try to create a page using `md2cf` v1.0.4:
```
admrfinch:devops:tmp$md2cf -s '~rfinch' DEMO.md
Creating new page: Bug Demo
Uploading attachments for page: Bug Demo
Uploading file: charizard-pokemon.png
405 Client Error:  for url: https://confluence.idicore.com/rest/api/content/358259237/child/attachment?allowDuplicated=true - b''
```
after downgrading to 1.0.3:
```
$ md2cf -s '~rfinch' DEMO.md
Updating page: Bug Demo
Uploading attachments for page: Bug Demo
Uploading file: charizard-pokemon.png
$ cat DEMO.md
# Bug Demo

![](charizard-pokemon.png)
```

**Comment by finchr** [comment]
any updates on this?

**Comment by iamjackg** [comment]
Hey, sorry about the delay. This seems to be due to an API difference between Confluence Cloud and Confluence Server.

#16 changed the method from POST to PUT in order to allow for attachment updates, but it looks like that only works for [Confluence Cloud](https://developer.atlassian.com/cloud/confluence/rest/api-group-content---attachments/#api-wiki-rest-api-content-id-child-attachment-put), whereas [Confluence Server](https://docs.atlassian.com/ConfluenceServer/rest/7.15.0/#api/content/{id}/child/attachment) needs a different POST request to update an attachment. @timothybonci FYI.

I'll take a look when I have a moment and see how best to manage this. It should be enough to check for the attachment beforehand and update it if necessary.

**Comment by timothybonci** [comment]
Yeah, that's a bit of a bummer.  It looks like both support POST /wiki/rest/api/content/{id}/child/attachment/{attachmentId}/data
So this would require an extra call to get the attachmentId first I guess, instead of being able to blindly PUT.

**Comment by iamjackg** [comment]
Hey @finchr, this should be fixed in `v1.1.0`. Could you please confirm? Thanks!

**Comment by iamjackg** [comment]
I'll assume this is fixed since I had no issues when testing it again :) Cheers!

**Comment by finchr** [comment]
> Hey @finchr, this should be fixed in `v1.1.0`. Could you please confirm? Thanks!

Hi @iamjackg the fix is confirmed! It's working with the latest version!
